llama.cpp

```diff
@@ -3722,16 +3729,23 @@ static void llm_load_tensors(
         model.tensors_by_name.emplace_back(ggml_get_name(cur), cur);
     }
 
-    ml.load_all_data(ctx, progress_callback, progress_callback_user_data, buf_mmap, use_mlock ? &model.mlock_mmap : NULL);
+    ok = ok && ml.load_all_data(ctx, progress_callback, progress_callback_user_data, buf_mmap, use_mlock ? &model.mlock_mmap : NULL);
+    if (progress_callback) {
+        // Even though the model is done loading, we still honor
+        // cancellation since we need to free allocations.
+        ok = ok && progress_callback(1.0f, progress_callback_user_data);
```

Comment:
The last `progress_callback` call is done in `load_all_data` now.

Reply:
I missed that. However, that's slightly different behavior. I wanted to make sure that if the loading is aborted even at `1.0f` that it still aborts.